pyneuroml/swc/ExportNML.py

```diff
@@ -0,0 +1,501 @@
+                self.point_indices_vs_seg_ids[this_point.id] = seg_id
+
+                segment = [f'      <segment id="{seg_id}" name="Seg_{seg_id}"']
+                if version.startswith("1"):
+                    segment.append(f' cable="{cable_id}">')
+                else:
+                    segment.append(f'> <!-- "Cable" is {cable_id}-->')
+
+                self.point_append(soma_points[1], 0.0, "proximal", segment)
+                self.point_append(soma_points[0], 0.0, "distal", segment)
+                segment.append("      </segment>")
+                self.segment_content.extend(segment)
+
+                # Create second segment (from second point to third point)
+                seg_id = self.next_segment_id
+                self.next_segment_id += 1
+                self.point_indices_vs_seg_ids[soma_points[2].id] = seg_id
+
+                segment = [f'      <segment id="{seg_id}" name="Seg_{seg_id}"']
+                if version.startswith("1"):
+                    segment.append(f' cable="{cable_id}">')
+                else:
+                    segment.append(f'> <!-- "Cable" is {cable_id}-->')
+
+                segment.append(f'        <parent segment="{seg_id-1}"/>')
+                self.point_append(soma_points[2], 0.0, "distal", segment)
+                segment.append("      </segment>")
+                self.segment_content.extend(segment)
+            elif (
+                this_point.id == soma_points[0].id or this_point.id == soma_points[2].id
+            ):
+                # Do nothing for first and third points as they're handled with the middle point
+                pass
+        elif new_cell and parent_point.type == SWCNode.SOMA:
+            print("First point of multi-point soma")
+            self.segment_content.append(
+                "      <!-- First point is of a multi point soma => not spherical! -->"
+            )
+        else:
+            print(f"Creating regular segment for soma point {this_point.id}")
+            self.create_segment(this_point, parent_point, cable_id, version, True)
+
+    def create_segment(
+        self,
+        this_point: SWCNode,
+        parent_point: SWCNode,
+        cable_id: int,
+        version: str,
+        new_cable: bool,
+    ):
+        """
+        Create a segment in the NeuroML structure.
+
+        :param this_point: The current SWCNode.
+        :param parent_point: The parent SWCNode.
+        :param cable_id: The ID of the current cable.
+        :param version: The NeuroML version being used.
+        :param new_cable: Whether this point starts a new cable.
+        """
+        print(
+            f"Creating segment: Point {this_point.id}, Parent {parent_point.id}, Cable {cable_id}"
+        )
+        seg_id = self.next_segment_id
+        self.next_segment_id += 1
+        self.point_indices_vs_seg_ids[this_point.id] = seg_id
+
+        segment = [f'      <segment id="{seg_id}" name="Seg_{seg_id}"']
```

Comment:
Now that this is all neuroml specific, is there a reason we're not creating objects using the libneuroml API? (`Segment` and `SegmentGroup`)? It's usually better than creating xml by hand.

Reply:
Can I get some more ideas on how to proceed with this? I am bit confused about using the libneuroml API.

Reply:
Sure, see this example for a start:

https://docs.neuroml.org/Userdocs/MultiCompartmentOLMexample.html#declaring-the-cell

When users use NeuroML, they don't write XML---that's tedious and error prone. Instead, they use the libNeuroML API:

```
cell = neuroml.Cell(...)
cell.add_segment(...)

..
```

it's basically a Python object model of the NeuroML schema, so a Cell contains Morphology, which contains Segments and SegmentGroups.

That's a better way of generating the NeuroML than writing XML by hand.

Let me know if this isn't clear and we can have a chat where I can explain it to you more.

Reply:
Thankyou I understand it now...will try to implement this

Reply:
Do take a look at the various "utility functions" in the Cell class, they should help with lots of things (hopefully):

https://libneuroml.readthedocs.io/en/latest/userdocs/coreclasses.html#cell